### PR TITLE
Fixes #33826 - Better determine if the host has a BMC interface

### PR DESCRIPTION
**Pull request by nbarrientos**
Using instead the unredacted password of the BMC interface.

See issue for more details :)

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
This makes sense to me but it'd be great if we had some tests for this.

**Comment by nbarrientos** [comment]
Thanks for looking!

> it'd be great if we had some tests for this

The situation when the setting is set to `true` (default) was [already covered by tests](https://github.com/theforeman/foreman/blob/02a057b02fdad42f4e6d867d1577f1f4b07f40c4/test/models/nics/bmc_test.rb#L13), I think. I've added an extra test case now to make sure that the BMC is still available if the setting is set to `false`, which is precisely what this patch is aim to fix.

**Review by ekohl — DISMISSED** [review]

**Comment by nbarrientos** [comment]
@ekohl It should be okay now. One element of the test matrix is red because the runner ran out of space I believe.

**Comment by nbarrientos** [comment]
Perhaps it's not worth anymore to merge this, see https://projects.theforeman.org/issues/33839.

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -724,7 +724,7 @@ def bmc_proxy
   def bmc_available?
     ipmi = bmc_nic
     return false if ipmi.nil?
-    (ipmi.password.present? && ipmi.username.present? && %w(IPMI Redfish).include?(ipmi.provider)) || ipmi.provider == 'SSH'
+    (ipmi.password_unredacted.present? && ipmi.username.present? && %w(IPMI Redfish).include?(ipmi.provider)) || ipmi.provider == 'SSH'
```
Comment (lzap) [inline app/models/host/managed.rb:727]: This could be something like `password_present?` but I am fine with this.

**Review by lzap — APPROVED** [review]
I am good, leaving @ekohl to merge this. Thanks.

**Review by ezr-ondrej — APPROVED** [review]
Thanks @nbarrientos :+1: 
Merging as ekohl will not get to this soon.

**Pull request merged**